src/tags/object/Video/Video.js

```diff
@@ -107,121 +103,72 @@ const Model = types
 
-      triggerSyncPause() {
-        // Audio v3
-        if (isFFDev2715) {
-          if (self.syncedObject) {
-            Super.triggerSyncPause();
-          } else {
-            self.handleSyncPause();
-          }
-        }
-        // Audio v1,v2
-        else {
-          Super.triggerSyncPause();
-        }
-      },
+    registerSyncHandlers() {
+      ['play', 'pause', 'seek'].forEach(event => {
+        self.syncHandlers.set(event, self.handleSync);
+      });
+      // self.syncHandlers.set('speed', self.handleSyncSpeed);
+    },
 
-      handleSyncSeek(time) {
-        // Audio v3
-        if (isFFDev2715) {
-          if (self.syncedDuration && time >= self.syncedDuration) {
-            self.ref.current.currentTime = self.ref.current.duration;
-          } else if (self.ref.current && !isTimeRelativelySimilar(self.ref.current.currentTime, time, self.ref.current.duration)) {
-            self.ref.current.currentTime = time;
-          }
-        }
-        // Audio v2,v1
-        else {
-          if (self.ref.current) {
-            self.ref.current.currentTime = time;
-          }
-        }
-      },
+    handleSync(data) {
+      if (!self.ref.current) return;
 
-      handleSyncPlay() {
-        // Audio v3
-        if (isFFDev2715) {
-          if (!self.isCurrentlyPlaying) {
-            self.isCurrentlyPlaying = true;
-            try {
-              self.ref.current?.play();
-            } catch {
-              // do nothing, just ignore the DomException
-              // just in case the video was in the midst of syncing
-            }
-          }
-        }
-        // Audio v2,v1
-        else {
-          self.ref.current?.play();
-        }
-      },
+      const video = self.ref.current;
 
-      handleSyncPause() {
-        // Audio v3
-        if (isFFDev2715) {
-          if (self.isCurrentlyPlaying) {
-            self.isCurrentlyPlaying = false;
-            try {
-              self.ref.current?.pause();
-            } catch {
-              // do nothing, just ignore the DomException
-              // just in case the video was in the midst of syncing
-            }
-          }
-        }
-        // Audio v2,v1
-        else {
-          self.ref.current?.pause();
-        }
-      },
+      if (data.playing) {
+        if (!video.playing) video.play();
+      } else {
+        if (video.playing) video.pause();
+      }
 
-      handleSyncDuration(duration) {
-        if (!isFFDev2715) return;
-        if (self.ref.current) {
-          self.setLength(duration * self.framerate);
-        }
-      },
+      video.currentTime = data.time;
+    },
+
+    handleSyncSpeed(speed) {
+      self.speed = speed;
+    },
+  }))
+  .actions(self => {
+    return {
+      afterCreate() {
+        const framerate = Number(parseValue(self.framerate, self.store.task?.dataObj));
 
-      handleSyncSpeed(speed) {
-        self.speed = speed;
+        if (!framerate || isNaN(framerate)) self.framerate = '24';
```

Comment:
Why are we keeping framerate as a String?

Reply:
Sadly that's our current way to deal with tag attrs =(
They are always strings (except for booleans), so we have to convert them when we use them. This can be actually done in other ways, but that requires to fix some places in whole app, so not for now...